server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -384,8 +384,9 @@ private void sendVerifyShardBeforeCloseRequest(final IndexShardRoutingTable shar
                 return;
             }
             final TaskId parentTaskId = new TaskId(clusterService.localNode().getId(), request.taskId());
+            final String syncId = UUIDs.randomBase64UUID();
```

Comment:
I like the overall PR. I was just wondering if it would make it sense to reuse the sync id from the primary if it had one. That could make sure not to destroy / juggle markers if they were already in place?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/41400/commits/9ed30cda0c143cfdb5b67343a26237537793f7d8.